DQM/L1TMonitor/interface/L1TStage2CaloLayer1.h

```diff
@@ -77,19 +80,124 @@ namespace ComparisonHelper {
+    dqm::reco::MonitorElement *ecalOccSentAndRecd_;
+    dqm::reco::MonitorElement *ecalOccSentFgVB_;
+    dqm::reco::MonitorElement *ecalOccSent_;
+    dqm::reco::MonitorElement *ecalOccTowerMasked_;
+    dqm::reco::MonitorElement *ecalTPRawEtCorrelation_;
+    dqm::reco::MonitorElement *ecalTPRawEtDiffNoMatch_;
+    dqm::reco::MonitorElement *ecalTPRawEtRecd_;
+    dqm::reco::MonitorElement *ecalTPRawEtSentAndRecd_;
+    dqm::reco::MonitorElement *ecalTPRawEtSent_;
+
+    dqm::reco::MonitorElement *ecalOccSentNotRecd_;
+    dqm::reco::MonitorElement *ecalOccRecdNotSent_;
+    dqm::reco::MonitorElement *ecalOccNoMatch_;
+
+    dqm::reco::MonitorElement *hcalOccEtDiscrepancy_;
+    dqm::reco::MonitorElement *hcalOccFbDiscrepancy_;
+    dqm::reco::MonitorElement *hcalOccFb2Discrepancy_;
+    dqm::reco::MonitorElement *hcalOccLinkMasked_;
+    dqm::reco::MonitorElement *hcalOccRecdEtWgt_;
+    dqm::reco::MonitorElement *hcalOccRecdFb_;
+    dqm::reco::MonitorElement *hcalOccRecdFb2_;
+    dqm::reco::MonitorElement *hcalOccSentAndRecd_;
+    dqm::reco::MonitorElement *hcalOccSentFb_;
+    dqm::reco::MonitorElement *hcalOccSentFb2_;
+    dqm::reco::MonitorElement *hcalOccSent_;
+    dqm::reco::MonitorElement *hcalOccTowerMasked_;
+    dqm::reco::MonitorElement *hcalTPRawEtCorrelationHBHE_;
+    dqm::reco::MonitorElement *hcalTPRawEtCorrelationHF_;
+    dqm::reco::MonitorElement *hcalTPRawEtDiffNoMatch_;
+    dqm::reco::MonitorElement *hcalTPRawEtRecd_;
+    dqm::reco::MonitorElement *hcalTPRawEtSentAndRecd_;
+    dqm::reco::MonitorElement *hcalTPRawEtSent_;
+
+    dqm::reco::MonitorElement *hcalOccSentNotRecd_;
+    dqm::reco::MonitorElement *hcalOccRecdNotSent_;
+    dqm::reco::MonitorElement *hcalOccNoMatch_;
+
+    dqm::reco::MonitorElement *ECALmismatchesPerBx_;
+    dqm::reco::MonitorElement *HBHEmismatchesPerBx_;
+    dqm::reco::MonitorElement *HFmismatchesPerBx_;
+
+    dqm::reco::MonitorElement *bxidErrors_;
+    dqm::reco::MonitorElement *l1idErrors_;
+    dqm::reco::MonitorElement *orbitErrors_;
+
+    dqm::reco::MonitorElement *ecalLinkErrorByLumi_;
+    dqm::reco::MonitorElement *ecalMismatchByLumi_;
+    dqm::reco::MonitorElement *hcalLinkErrorByLumi_;
+    dqm::reco::MonitorElement *hcalMismatchByLumi_;
+
+    dqm::reco::MonitorElement *maxEvtLinkErrorsByLumiECAL_;
+    dqm::reco::MonitorElement *maxEvtLinkErrorsByLumiHCAL_;
+    dqm::reco::MonitorElement *maxEvtLinkErrorsByLumi_;
+
+    dqm::reco::MonitorElement *maxEvtMismatchByLumiECAL_;
+    dqm::reco::MonitorElement *maxEvtMismatchByLumiHCAL_;
+    dqm::reco::MonitorElement *maxEvtMismatchByLumi_;
+
+    dqm::reco::MonitorElement *last20Mismatches_;
+
+    //these maps store the maximum number of evt and link mismatches per lumi section.
+    //they are read back at the end of runs
+    //Made std::unique_ptr's for better memory management
+    std::unique_ptr<std::map<std::string, int>> maxEvtLinkErrorsMapECAL{std::make_unique<std::map<std::string, int>>()};
```

Comment:
How is `unique_ptr`  better here than
```cpp
std::map<std::string, int> maxEvtLinkErrorsMapECAL;
```
?

Reply:
I believe the reasoning behind this was, as you were noting above, my way to dodge the compiler errors for a `const` function modifying class elements by making the class element a `const` pointer to some mutable object that I was working with (which, I grant you, seems like a textbook way to write terminally buggy code). 

I think continuing from the above, what is necessary for this module to maintain it's previous functionality and also is a thread safe dynamically sized vector/map type of object whose contents are mutable in the analysis functions for storing luminosity block information at the run cached level. Will `tbb::concurrent_vector<>`'s provide this?